### ISSUE TITLE
entry arrays are concatenated in reverse order of arguments

### ISSUE THREAD

**Issue by jedwards1211** [survivejs/webpack-merge#10]
This caught me off guard:
```js
> merge({entry: ['entry1']}, {entry: ['entry2']});
{ entry: [ 'entry2', 'entry1' ] }
```
Is there any particular reason for that?  Seems to me the opposite order is less surprising.

**Comment by bebraw** [comment]
Hi,

The order was inherited from loaders (evaluated from right to left).

Maybe we should do an exception for `entry`?

**Comment by jedwards1211** [comment]
Hmm.  The README seems conflicting about that though.  According to it, when the loaders are `'babel'` and `['coffee']` `merge.smart` merges them to `['babel', 'coffee']`, yet when the loaders are `['babel']` and `['coffee']` they get merged as `['coffee', 'babel']`.  Is that correct?

It's not really clear to me which order is desirable for loader merging anyway, are there any real-world examples of doing that? (does anyone actually use coffee and babel together?)

I think at least for anything besides loaders left-to-right concatenation order is less surprising (and I do hope the assignment order for objects is left to right!)

**Comment by bebraw** [comment]
> Hmm. The README seems conflicting about that though. According to it, when the loaders are 'babel' (not in an array) and ['coffee'] merge.smart merges them to ['babel', 'coffee'], yet when the loaders are ['babel'] (in an array) and ['coffee'] they get merged as ['coffee', 'babel']. Is that correct?

Thanks for bringing that up, I fixed the README example. Should have been `['babel', 'coffee']`.

> It's not really clear to me which order is desirable for loader merging anyway, are there any real-world examples of doing that? (does anyone actually use coffee and babel together?)

To quote change log:

> Breaking: Change smart merging behavior for loaders field so that it prepends loaders instead of appending them. The benefit of this is that now it's possible to specialize loader setup in a predictable manner. For example you can have a linter set up at the root and expect it to become evaluated first always.

This way it's easier to reason what's going on with loaders. You can count on "top to bottom" order.

> I think at least for anything besides loaders left-to-right concatenation order is less surprising (and since it appears object property assignments are processed left-to-right as expected)

It's possible doing prepend for everything is too much. What if we restricted it just to `preLoaders`, `loaders`, and `postLoaders`? After this you get appending behavior for everything else and the system remains logical. Now the `entry` order can be surprising indeed.

This problem would have been far easier if webpack evaluated loaders from left to right and top to bottom. :smile:

**Comment by jedwards1211** [comment]
Yeah that restriction sounds good to me.

Webpack loaders are evaluated bottom to top?? I couldn't find anything in the configuration documentation about that...

**Comment by bebraw** [comment]
Some of the bits are a bit poorly documented, yeah. It should be fairly easy to verify this, though.

I'll make this change to the next version. Thanks for bringing up the issue!

**Comment by jedwards1211** [comment]
Thank you, this library is already really helpful!

**Comment by bebraw** [comment]
Fixed in `0.8.0`. Now prepend logic is used only with loaders. More logical this way. :+1:

**Comment by jedwards1211** [comment]
Very cool, thank you!
